tests/test_spatial_tools.py

```diff
@@ -53,3 +58,57 @@ def test_merge_rasters():
     diff = dem.data - merged_dem.data
 
     assert np.abs(np.nanmean(diff)) < 0.05
+
+
+def test_hillshade():
+    """Test the hillshade algorithm, partly by comparing it to the GDAL hillshade function."""
+    warnings.simplefilter("error")
+
+    def make_gdal_hillshade(filepath) -> np.ndarray:
+        temp_dir = tempfile.TemporaryDirectory()
+        temp_hillshade_path = os.path.join(temp_dir.name, "hillshade.tif")
+        gdal_commands = ["gdaldem", "hillshade",
```

Comment:
Note that big of a deal, but you can call this directly in Python with gdal.DEMProcessing !

Reply:
That may actually be a big deal, because the GitHub Actions CI don't have GDAL in their PATH and I don't know how to fix this, hence the ugly "only test if gdal is found" assertion in the test.